docs/configuring_qmk.md

```diff
@@ -6,113 +6,107 @@ sidebar_label: Configuring QMK
+
+For acceptance into the upstream QMK repository this folder must be named `via`, but that is not an absolute requirement for your copy of the repository. It is possible to add VIA support to any keymap with the correct configuration.
 
 ## Create a `rules.mk` in `keyboards/<keyboard_name>/keymaps/via`
 
-In most cases this will only require:
+In most cases this file will only require:
 
     VIA_ENABLE = yes
 
 Make sure `yes` is lowercase. `YES` will not enable VIA and you will be most confused.
 
 This will enable dynamic keymaps, raw HID message handling and bootmagic lite.
 
-Note: ‘bootmagic lite’ is highly recommended to the point of being essential. ‘bootmagic lite’ is the ability to hold down Esc (or some other key) while inserting USB cable to both jump to bootloader and reset the EEPROM. Thus if for some reason the EEPROM contains data that is out of sync with the firmware, and things aren’t working as expected (e.g. garbage keycodes in VIA), the device can be ‘factory reset’. It also makes redundant the need for a ‘RESET’ keycode in the keymap. The VIA implementation in QMK overrides the default ‘bootmagic_lite()’ but the only difference is additionally invalidating VIA’s EEPROM memory.
+‘Bootmagic Lite’ is the ability to hold down Esc (or some other key) while plugging in the keyboard to both jump to bootloader and reset the EEPROM. Thus if for some reason the EEPROM contains data that is out of sync with the firmware, and things aren’t working as expected (e.g. garbage keycodes in VIA), the device can be ‘factory reset’. It also makes redundant the need for a ‘QK_BOOT’ keycode in the keymap.
 
-The keyboard’s own `rules.mk` should be compatible with the VIA-specific firmware and so nothing else is needed.
+Do not put `VIA_ENABLE = yes` in the keyboard directory’s `rules.mk`. This configuration should only be added to VIA-specific keymaps, not enabled by default at the keyboard level.
 
-Currently, VIA is incompatible with features that change the integer values of `enum quantum_keycodes`, i.e. that optionally add enum values to `enum quantum_keycodes` and change the sequential assignment of integer values to enum names. As such, the following features must be disabled for VIA support, until this issue is fixed (i.e. a refactor of `enum quantum_keycodes`):
+## Create a `keymap.c` or `keymap.json` in `keyboards/<keyboardname>/keymaps/via`
 
-- LEADER_ENABLE
-- FAUXCLICKY_ENABLE
-- MIDI_ENABLE
-- BLUETOOTH_ENABLE
-- KEY_LOCK_ENABLE
-- TERMINAL_ENABLE
+The keymap in the `via` keymap folder should use a `LAYOUT_*()` macro that allows all the electrical positions to be mapped, even if that layout isn't physically possible.
 
-## Create a `keymap.c` in `keyboards/<keyboardname>/keymaps/via`
+By default, dynamic keymaps have 4 layers. These will be automatically populated with `KC_TRNS` keycodes as necessary, so there is no need to create additional blank layers in your keymap.
```

Comment:
Perhaps something like this? It seems to suggest you need the base + 4 empty layers.
```suggestion
By default, dynamic keymaps have 4 layers. These will be automatically populated with `KC_TRNS` keycodes as necessary, so there is no need to create more than 4 layers in your keymap in the default case.
```